frontend/simulation_worker/src/lib.rs

```diff
@@ -60,9 +60,10 @@ impl yew_agent::Worker for SimAgent {
                 self.link.respond(who, Response::Snapshot { snapshot });
             }
             Request::Snapshot { ticks, nonce } => {
-                if self.errored {
+                if self.errored || self.sim().status() != Status::Running {
```

Comment:
This keeps responding with snapshots after the end of the simulation if we don't guard against it.